azurerm/internal/services/mssql/mssql_server_transparent_data_encryption_resource.go

```diff
@@ -0,0 +1,244 @@
+		}
+	}
+
+	encryptionProtectorObject := sql.EncryptionProtector{
+		EncryptionProtectorProperties: &encryptionProtectorProperties,
+	}
+
+	futureEncryptionProtector, err := encryptionProtectorClient.CreateOrUpdate(ctx, serverId.ResourceGroup, serverId.Name, encryptionProtectorObject)
+	if err != nil {
+		return err
+	}
+
+	if err = futureEncryptionProtector.WaitForCompletionRef(ctx, encryptionProtectorClient.Client); err != nil {
+		return fmt.Errorf("waiting on create/update future for Encryption Protector on server %q (Resource Group %q): %+v", serverId.Name, serverId.ResourceGroup, err)
+	}
+
+	// Encryption protector always uses "current" for the name
+	id := parse.NewEncryptionProtectorID(serverId.SubscriptionId, serverId.ResourceGroup, serverId.Name, "current")
+	d.SetId(id.ID())
+
+	return resourceMsSqlTransparentDataEncryptionRead(d, meta)
+}
+
+func resourceMsSqlTransparentDataEncryptionRead(d *schema.ResourceData, meta interface{}) error {
+	encryptionProtectorClient := meta.(*clients.Client).MSSQL.EncryptionProtectorClient
+
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.EncryptionProtectorID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := encryptionProtectorClient.Get(ctx, id.ResourceGroup, id.ServerName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error making Read request for Encryption Protector on server %s: %v", id.ServerName, err)
+	}
+
+	serverId := parse.NewServerID(id.SubscriptionId, id.ResourceGroup, id.ServerName)
+	d.Set("server_id", serverId.ID())
+
+	log.Printf("[INFO] Encryption protector key type is %s", resp.EncryptionProtectorProperties.ServerKeyType)
+
+	keyVaultKeyId := ""
+
+	// Only set the key type if it's an AKV key. For service managed, we can omit the setting the key_vault_key_id
+	if resp.EncryptionProtectorProperties != nil && resp.EncryptionProtectorProperties.ServerKeyType == sql.AzureKeyVault {
+		log.Printf("[INFO] Setting Key Vault URI to %s", *resp.EncryptionProtectorProperties.URI)
+
+		keyVaultKeyId = *resp.EncryptionProtectorProperties.URI
+	}
+
+	if err := d.Set("key_vault_key_id", keyVaultKeyId); err != nil {
+		return fmt.Errorf("setting key_vault_key_id`: %+v", err)
+	}
+
+	return nil
+}
+
+func resourceMsSqlTransparentDataEncryptionDelete(d *schema.ResourceData, meta interface{}) error {
+	// Note that encryption protector cannot be deleted. It can only be updated between AzureKeyVault
+	// and SystemManaged. For safety, when this resource is deleted, we're resetting the key type
+	// to service managed to prevent accidental lockout if someone were to delete the keys from key vault
+
+	encryptionProtectorClient := meta.(*clients.Client).MSSQL.EncryptionProtectorClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	serverId, err := parse.ServerID(d.Get("server_id").(string))
```

Comment:
Although using a ServerID works here, could we use an EncryptionProtectorID instead?

Reply:
I could be wrong here, but I don't think it'll work in this case. The schema input being defined is the server_id (which takes the format: /sub/rg/serverName) not an encryption protector ID. As a result, when doing the delete it doesn't have enough information to parse an encryption protector id (which takes the format /sub/rg/serverName/encryptionProtectorName). 

The reason we don't need the encryption protector ID and instead rely on the server ID is because the EP is effectively a singleton, and there can only be one assigned to a server and the name is fixed (API sets it to "current").

Reply:
If you meant, `id, err := parse.EncryptionProtectorID(d.Id())` then yes I can make the adjustment. I forgot about the d.Id() property :)

Reply:
Sorry for the ambiguity - that is indeed what I was trying to say :D